homeassistant/components/telegram_bot/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""
+Component to receive telegram messages.
+
+Either by polling or webhook.
+"""
+
+import asyncio
+import logging
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_PLATFORM, CONF_API_KEY
+
+from homeassistant.helpers.entity_component import EntityComponent
+
+DOMAIN = 'telegram_bot'
+
+_LOGGER = logging.getLogger(__name__)
+
+EVENT_TELEGRAM_COMMAND = 'telegram_command'
+EVENT_TELEGRAM_TEXT = 'telegram_text'
+
+ATTR_COMMAND = 'command'
+ATTR_USER_ID = 'user_id'
+ATTR_ARGS = 'args'
+ATTR_FROM_FIRST = 'from_first'
+ATTR_FROM_LAST = 'from_last'
+ATTR_TEXT = 'text'
+
+CONF_ALLOWED_CHAT_IDS = 'allowed_chat_ids'
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): cv.string,
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_ALLOWED_CHAT_IDS): vol.All(cv.ensure_list,
+                                                 [cv.positive_int])
+}, extra=vol.ALLOW_EXTRA)
+
+
+class WrongMessageException(Exception):
+    """Exception to raise when an incoming message has wrong format."""
+
+    pass
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Track states and offer events for sensors."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
```

Comment:
Please do not use the EntityComponent but use the few building blocks that the EntityComponent uses that you need. See the notify component for an example: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/notify/__init__.py#L86-L112